### PR TITLE
Login: Fix fatal error when navigating from reset password page

### LINKED ISSUE #20663: Having been password reset user is redirected to page with error
**What happened**:
Having reset password user is redirected to the page where error occurs. [Screenshot](https://imgur.com/a/zdFqCzr)

**What you expected to happen**:
User should be redirected to login page.

**How to reproduce it (as minimally and precisely as possible)**:
We figured out that the error appears exactly after Grafana v.6.4.2. v.6.4.1 doesn't reproduce this kind of error.

**Anything else we need to know?**:
It doesn't depend on browser.

**Environment**:
- Grafana version: 6.4.2
- Data source type & version: -
- OS Grafana is installed on: Ubuntu 18.04.3
- User OS & Browser: Windows 10, Firefox 70.0.1
- Grafana plugins: 
- Others:

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Fixes an issue with the reset password page when changing password

**Which issue(s) this PR fixes**:
Fixes #20663

**Special notes for your reviewer**:
To test this you need to setup smtp locally and send a reset password link.

### CODE REVIEW

File: public/app/partials/reset_password.html [hunk 1]
```diff
@@ -10,7 +10,7 @@ <h3 class="p-b-1">Reset password</h3>
 		<div ng-if="disableLoginForm">
 			You cannot reset password when login form is disabled.
 		</div>
-		<form name="sendResetForm" class="login-form gf-form-group" ng-show="mode === 'send'" ng-hide="ldapEnabled || authProxyEnabled || disableLoginForm">
+		<form name="sendResetForm" class="login-form gf-form-group" ng-show="mode === 'send'" ng-hide="ldapEnabled || authProxyEnabled || disableLoginForm || mode === 'reset'">
 			<div class="gf-form">
 					<span class="gf-form-label width-7">User</span>
```
Comment (dprokop) [inline public/app/partials/reset_password.html]: Hm, so when is the Username form used at all?

File: public/app/core/components/Login/LoginServiceButtons.tsx [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
+
+  return {
+    saml: {
+      enabled: config.samlEnabled,
+      name: 'SAML',
+      className: 'github',
+      icon: 'key',
+    },
+    google: {
+      enabled: oauthEnabled && config.oauth.google,
```
Comment (dprokop) [inline public/app/core/components/Login/LoginServiceButtons.tsx]: Nice catch with the oauthEnabled  - did you run into some problems with it?

**Review by dprokop — APPROVED** [review]
LGTM

File: public/app/partials/reset_password.html (hunk 1 above)
Comment (peterholmberg) [reply public/app/partials/reset_password.html]: When you click "Forgot password" on login page.

File: public/app/core/components/Login/LoginServiceButtons.tsx (hunk 2 above)
Comment (peterholmberg) [reply public/app/core/components/Login/LoginServiceButtons.tsx]: Yeah, this is what caused the bug. If oauth was undefined we got the error in the issue

**Comment by torkelo** [comment]
moving this to 6.5.2 patch release milestone as it seems like an important bug fix

**Pull request merged**